src/services/pagoPAProxyService.ts

```diff
@@ -105,3 +121,35 @@ export default class PagoPAProxyService {
     }
   }
 }
+
+/**
+ * Interface used to check if an http response has a custom title
+ */
+const ErrorBodyWithTitle = t.interface({
+  parsedBody: t.interface({
+    title: t.string
+  })
+});
+type ErrorBodyWithTitle = t.TypeOf<typeof ErrorBodyWithTitle>;
+
+/**
+ * Determine the error message to send to CDApp using a whitelist or a generic internal message
+ * Check Error Mapping documentation at:
+ * https://docs.google.com/document/d/1Qqe6mSfon-blHzc-ldeEHmzIkVaElKY5LtDnKiLbk80/edit#
+ */
+export function getErrorMessageForApp(
+  httpOperationResponse: HttpOperationResponse | undefined
+): string {
```

Comment:
```ts
ErrorBodyWithTitle.decode(httpOperationResponse)
  .map(_ => _.parsedBody.title)
  .filterOrElse(_ => pagoPaErrorWhiteList.indexOf(_) !== -1, internalErrorMessage)
```

Reply:
https://github.com/gcanti/fp-ts/blob/master/src/Either.ts#L144

Reply:
filterOrElse returns a Right<string> or a Left<ValidationError>

In this case I would like to return a string in both case (just select the string decoded or a default string).
To use filterOrElse I should pass a ValidationError as second parameter (internalErrorMessage is not allowed directly) and store the result to a const. Then, to extract again the value from this const.

Reply:
then this:

```ts
ErrorBodyWithTitle.decode(httpOperationResponse)
  .map(_ => _.parsedBody.title)
  .filterOrElse(_ => pagoPaErrorWhiteList.indexOf(_) !== -1, internalErrorMessage)
  .getOrElse(internalErrorMessage)
```

Reply:
[ts] Argument of type '"Internal error."' is not assignable to parameter of type 'ValidationError[]'.

I can't pass internalErrorMessage as second parameter to filterOrElse. As I wrote in the last message, it requires ValidationError.